src/settings/target/template_config.rs

```diff
@@ -0,0 +1,112 @@
+use std::collections::HashMap;
+
+use serde::{Deserialize, Serialize};
+
+use crate::terminal::message;
+
+use super::{Environment, KvNamespace, Site, TargetType};
+
+#[derive(Clone, Debug, Default, Deserialize, Serialize)]
+pub struct TemplateConfig {
```

Comment:
why make a new struct for this rather than using manifest?

Reply:
oh. that's a really good point. initially i introduced this new struct because it was only taking in `type` and `webpack_config` but now since we're taking everything it makes sense to use `Manifest`

Reply:
ah. makes sense. yah i think a lot of logic might be simplified if the logic was just "pull manifest and overwrite some 💩 "

Reply:
Oop - just kidding! spoke too soon! Everything in `TemplateConfig` is an `Option` so that if it's omitted from a given template, wrangler generate does not even give a care. We don't want everything to be an `Option` in `Manifest` though.

Reply:
alright. i think there are refactors we can do to dedup a lot of this stuff but they are not for this PR.

Reply:
I still think there is a way we can use Manifest in place of a new, duplicative TemplateConfig struct... (let's have our cake and eat it too!). There should be a serde option that allows something that isn't present to be deserialized as an empty string (check out https://serde.rs/field-attrs.html, specifically a variant of `#[serde(default)]`.